python/py-ply/Portfile

```diff
@@ -22,12 +22,16 @@ checksums           rmd160  d0039bbbf095aeb7ed82fc2fab4b4699e8406d50 \
                     sha256  00c7c1aaa88358b9c765b6d3000c6eec0ba42abca5351b095321aef446081da3 \
                     size    159130
 
-python.versions     27 35 36 37 38 39 310 311
+python.versions     27 35 36 37 38 39 310 311 312
 
 if {${name} ne ${subport}} {
     depends_build-append \
                     port:py${python.version}-setuptools \
 
+    if {${python.version} >= 312} {
+        patchfiles-append patch-unittest-assert.diff
+    }
```

Comment:
it seems to me from the Python unittest documentation that `self.assertTrue` can always be used instead of `self.assert_` that was indeed deprecated since Python 3.1 and finally removed in Python 3.12. 

Since this is a very simple change that can be applied unconditionally, I would suggest instead of adding a 800+ patchfile to do here instead:
```
    post-patch {
        reinplace "s|assert_|assertTrue|g" \
            ${worksrcpath}/test/testlex.py ${worksrcpath}/test/testyacc.py
    }
```

what do you think?

Reply:
The string-replace alternative will be fine. I did consider it, but a butterfly in the Amazon pushed me in the patch fold. We still have a Python 2.7 subport so it can't be unconditionally. Updated accordingly.

Reply:
who says that it doesn't work with PY27? That isn't clear from the unittest documentation and it worked fine when I tried. What error do you get when you run the tests with PY27 after applying the patch?

Reply:
You're right, the deprecation was also added in Python 2.7:
https://docs.python.org/2.7/library/unittest.html?highlight=unittest#deprecated-aliases.

Updated.